sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -1,164 +1,122 @@
-# Azure.Core public serialization samples
+# Azure.Core public serialization
 
-## Using explicit cast
+The .NET Azure SDK libraries support serialization and deserialization of most client library model types using the static `ModelSerializer` class. 
 
-When using protocol methods for advanced handling of RequestContext it is still possible to use the strongly typed models.
-There is an explicit cast operator that can be used to convert the protocol Response to the strongly typed model.
-There is also an explicit cast operator that can be used to convert the strongly typed model to the protocol RequestContent.
+## Key Concepts
 
-### Serialization
+- [Using the ModelSerializer](#using-the-modelserializer)
+- [Using models with protocol methods](#using-models-with-protocol-methods)
+- [Using JsonSerializer](#using-jsonserializer)
+- [Envelope bring your own model case](#envelope-bring-your-own-model-case)
 
-```C# Snippet:ExplicitCast_Serialize
-PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
-DogListProperty dog = new DogListProperty("myPet");
-Response response = client.CreatePet("myPet", (RequestContent)dog);
-var response2 = client.CreatePet("myPet", RequestContent.Create(dog));
-```
-
-### Deserialization
-
-```C# Snippet:ExplicitCast_Deserialize
-PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
-Response response = client.GetPet("myPet");
-DogListProperty dog = (DogListProperty)response;
-Console.WriteLine(dog.IsHungry);
-```
+## Using the ModelSerializer
+The `ModelSerializer` class enables users to serialize and deserialize any Azure models that implement the `IModelSerializable<T>` interface. The following example shows a Dog model being serialized and deserialized.
 
-Given that explicit cast does not allow for serialization options we might also consider a static `FromResponse` and instance `ToRequestContent` methods.
+```C# Snippet:BaseModelSerializer
+Dog doggo = new Dog
+{
+    Name = "Doggo",
+    Age = 7
+};
 
-## Using ModelSerializer for NewtonSoftJson
+BinaryData data = ModelSerializer.Serialize(doggo);
 
-By using the ModelSerializer class, a new instance of Dog does not need to be created before calling Deserialize. Also added ObjectSerializer to Options class so different kinds of Serializers can be used.
+Dog dog = ModelSerializer.Deserialize<Dog>(data);
+```
 
-### Serialization
+By default, the ModelSerializer gives you the Json representation of the model. Some services accept XML so if you want to send the serialized data to an Azure service that accepts XML, you can use the `ModelSerializerFormat` enum to specify the format. By using the `Wire` format, the serializer automatically picks what the service would use.
 
-```C# Snippet:NewtonSoft_Serialize
-DogListProperty dog = new DogListProperty
+```C# Snippet:ModelSerializerWithFormat
+Dog doggo = new Dog
```

Comment:
Krzysztof left a comment earlier that I agree with, but I think it disappeared.   We should consider using an actual model from one of the libraries - or at least a mock type that strongly resembles one.   `Dog` feels arbitrary and we don't define it anywhere, so I could not just copy/paste this snippet and see it work.